ideascube/management/commands/tags.py

```diff
@@ -121,3 +135,53 @@ def list(self, options):
         print(row.format('.' * 40, '.' * 40, '.' * 40))
         for tag in Tag.objects.order_by('slug'):
             print(row.format(tag.name, tag.slug, self._count(tag.name)))
+
+    def sanitize(self, options):
+        all_tags = list(Tag.objects.all().order_by('id').reverse())
+        while all_tags:
+            tag = all_tags.pop()
+            new_tag_name = sanitize_tag_name(tag.name)
+
+            if not new_tag_name:
+                # No need to delete relation, happy cascading !
+                tag.delete()
+                continue
+
+            other_equivalent_tags = {t for t in all_tags
+                if new_tag_name == sanitize_tag_name(t.name)}
```

Comment:
You're going to be doing that a lot (although `all_tags` is going to get smaller every time).

Why not do, before the `while` loop:

```python
        all_tags = ((sanitize_tag_name(t.name), t) for t in Tag.objects.all())
        all_tags = sorted(all_tags, key=itemgetter(0))
        all_tags = groupby(all_tags, key=itemgetter(0))
```

Then you can just iterate over `all_tags`:

```python
        for sanitized_name, tags in all_tags:
            if not sanitized_name:
                # Delete the tags
                continue

            # Keep the first tag as the one and only
            tags = list(tags)  # Or `tags = sorted(tags, key=attrgetter('id'))` if you really want to keep the first created one, but I don't think it matters at all
            tag_to_keep = tags[0]
            other_equivalent_tags = tags[1:]
            
            # Then do what you have to do with the `other_relations`, drop the `other_equivalent_tags` and rename the tag
```

Wouldn't that be simpler?